conformance/utils/kubernetes/helpers.go

```diff
@@ -356,27 +370,39 @@ func WaitForGatewayAddress(t *testing.T, client client.Client, timeoutConfig con
 	return net.JoinHostPort(ipAddr, port), waitErr
 }
 
-// GatewayListenersMustHaveCondition checks if every listener of the specified gateway has a
-// certain condition.
-func GatewayListenersMustHaveCondition(t *testing.T, client client.Client, timeoutConfig config.TimeoutConfig, gwName types.NamespacedName, condition metav1.Condition) {
+// GatewayListenersMustHaveConditions checks if every listener of the specified gateway has all
+// the specified conditions.
+func GatewayListenersMustHaveConditions(t *testing.T, client client.Client, timeoutConfig config.TimeoutConfig, gwName types.NamespacedName, conditions []metav1.Condition) {
 	t.Helper()
 
-	waitErr := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, timeoutConfig.GatewayListenersMustHaveCondition, true, func(ctx context.Context) (bool, error) {
-		var gw v1beta1.Gateway
-		if err := client.Get(ctx, gwName, &gw); err != nil {
-			return false, fmt.Errorf("error fetching Gateway: %w", err)
-		}
+	var wg sync.WaitGroup
+	wg.Add(len(conditions))
 
-		for _, listener := range gw.Status.Listeners {
-			if !findConditionInList(t, listener.Conditions, condition.Type, string(condition.Status), condition.Reason) {
-				return false, nil
-			}
-		}
+	for _, condition := range conditions {
+		go func(condition metav1.Condition) {
```

Comment:
any particular reason to do this concurrently rather than just loop through the desired conditions and validate all 3 are present in one go?

Reply:
Nothing special, honestly.
I guess I'll switch it to the way you suggested

Reply:
I also would prefer the more efficient option that @sunjayBhatia suggested, but don't think it needs to block this.

Reply:
Sure, I will follow up on it at issue #2459